ddmrp/models/stock_move.py

```diff
@@ -74,6 +79,24 @@ def _find_buffers_to_update_nfp(self):
             )
         return out_buffers, in_buffers
 
+    def _find_buffers_to_auto_procure(self):
+        # If a quantity is increased at a buffer replenishment location, this
+        # could release a new auto procure
+        in_buffers = self.env["stock.buffer"]
+        for move in self:
+            in_buffers |= move.mapped("product_id.buffer_ids").filtered(
+                lambda buffer: (
+                    buffer.distributed_source_location_id
+                    and not move.location_id.is_sublocation_of(
+                        buffer.distributed_source_location_id
+                    )
+                    and move.location_dest_id.is_sublocation_of(
+                        buffer.distributed_source_location_id
+                    )
+                )
```

Comment:
only cosmetic, I would extract that in a dedicated method. Would be easier to understand.